silhouette-cas/src/main/scala/com/mohiva/play/silhouette/impl/providers/cas/CASClient.scala

```diff
@@ -0,0 +1,57 @@
+package com.mohiva.play.silhouette.impl.providers.cas
+
+import org.jasig.cas.client.Protocol
+import org.jasig.cas.client.authentication.AttributePrincipal
+import org.jasig.cas.client.validation._
+import scala.concurrent.{ ExecutionContext, Future }
+import com.mohiva.play.silhouette.api.Logger
+
+object CASClient {
+  val SERVICE_TICKET_PARAMETER = "ticket"
+}
+
+case class CasProtocol(protocol: Protocol, ticketValidatorFactory: CASSettings => TicketValidator)
+
+object CasProtocols {
+
+  private def casValidatorWithEncoding[T <: AbstractUrlBasedTicketValidator](constructor: String => T) = {
```

Comment:
The order for variables and methods should be as follows:

- public variables
- protected variables
- private variables
- public methods
- protected methods
- private methods